snakemake/executors/slurm/slurm_jobstep.py

```diff
@@ -64,42 +64,45 @@ async def _wait_for_jobs(self):
     def run(self, job, callback=None, submit_callback=None, error_callback=None):
         jobsteps = dict()
 
-        # TODO revisit special handling for group job levels via srun at a later stage
-        # if job.is_group():
-
-        #     def get_call(level_job, aux=""):
-        #         # we need this calculation, because of srun's greediness and
-        #         # SLURM's limits: it is not able to limit the memory if we divide the job
-        #         # per CPU by itself.
-
-        #         level_mem = level_job.resources.get("mem_mb")
-        #         if isinstance(level_mem, TBDString):
-        #             level_mem = 100
-
-        #         mem_per_cpu = max(level_mem // level_job.threads, 100)
-        #         exec_job = self.format_job_exec(level_job)
-
-        #         # Note: The '--exlusive' flag is a prevention for triggered job steps within an allocation
-        #         #       to oversubscribe within a given c-group. As we are dealing only with smp software
-        #         #       the '--ntasks' is explicitly set to 1 by '-n1' per group job(step).
-        #         return (
-        #             f"srun -J {job.groupid} --jobid {self.jobid}"
-        #             f" --mem-per-cpu {mem_per_cpu} -c {level_job.threads}"
-        #             f" --exclusive -n 1 {aux} {exec_job}"
-        #         )
-
-        #     for level in list(job.toposorted):
-        #         # we need to ensure order - any:
-        #         level_list = list(level)
-        #         for level_job in level_list[:-1]:
-        #             jobsteps[level_job] = subprocess.Popen(
-        #                 get_call(level_job), shell=True
-        #             )
-        #         # now: the last one
-        #         # this way, we ensure that level jobs depending on the current level get started
-        #         jobsteps[level_list[-1]] = subprocess.Popen(
-        #             get_call(level_list[-1], aux="--dependency=singleton"), shell=True
-        #         )
+        if job.is_group():
+
+            def get_call(level_job, aux=""):
+                # we need this calculation, because of srun's greediness and
+                # SLURM's limits: it is not able to limit the memory if we divide the job
+                # per CPU by itself.
+
+                # check whether level_job.resources.mem_mb is TBD
+                # '100' shall be the (tiny) default, if no value has been specified
+                level_mem = (
+                    100
+                    if level_job.resources.mem_mb != type(int)
+                    else level_job.resources.mem_mb
+                )
```

Comment:
I think now I remember the problem again. This 100 can be completely wrong. mem_mb can be TBD if the memory depends on some input file that is created during execution of the group. If we just execute the entire group via Snakemake as it happened when I commented out the job.is_group() case, the job will be started with the correct mem_mb because Snakemake re-evaluates that.

Reply:
One way to fix this is to wait for the previous levels to finish whenever a TBD in mem_mb.